yaptide/routes/celery_routes.py

```diff
@@ -156,19 +159,19 @@ def delete(user: UserModel):
                                     })
 
         tasks = fetch_celery_tasks_by_sim_id(sim_id=simulation.id)
-
-        celery_ids = [task.celery_id for task in tasks]
-
-        result: dict = cancel_job(merge_id=simulation.merge_id, celery_ids=celery_ids)
-
-        if "merge" in result:
-            update_simulation_state(simulation=simulation, update_dict=result["merge"])
-            for i, task in enumerate(tasks):
-                update_task_state(task=task, update_dict=result["tasks"][i])
-
-            return yaptide_response(message="", code=200, content=result)
-
-        return error_internal_response()
+        celery_ids = [
+            task.celery_id for task in tasks
+            if task.task_state in [EntityState.PENDING.value, EntityState.RUNNING.value, EntityState.UNKNOWN.value]
+        ]
+        celery_app.control.revoke(simulation.merge_id, terminate=True, signal="SIGINT")
+        celery_app.control.revoke(celery_ids, terminate=True, signal="SIGINT")
```

Comment:
```suggestion
        celery_app.control.terminate(celery_ids, signal="SIGINT")
```

wouldn't that work as well ?

Reply:
Revoke is need here because it prevents queued tasks from running in future